projects/storefrontlib/src/cms-components/content/tab-paragraph-container/tab-paragraph-container.component.ts

```diff
@@ -1,41 +1,66 @@
-import { Component, ChangeDetectionStrategy } from '@angular/core';
-// import { AbstractProductComponent } from '../abstract-product-component';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ViewChild,
+  ElementRef,
+} from '@angular/core';
+import { CMSTabParagraphContainer } from '@spartacus/core';
+import { CmsComponentData } from 'projects/storefrontlib/src/cms-structure';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
 
 @Component({
   selector: 'cx-tab-paragraph-container',
   templateUrl: './tab-paragraph-container.component.html',
-  styleUrls: ['./tab-paragraph-container.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class TabParagraphContainerComponent {
-  // extends AbstractProductComponent {
-  // @ViewChild('tabContent') tabContent: Element;
-  // fetchData() {
-  //     // console.log(this.contextParameters.tab);
-  //     // this.contextParameters.tab.title = new BehaviorSubject<any>({});
-  //     // this.contextParameters.tab.title.next('test');
-  //     // this.contextParameters.tab.title.complete();
-  //     // const s = new AsyncSubject<any>();
-  //     // this.occProductSearchService.query(query)
-  //     //     .then((pageData) => {
-  //     //         s.next(pageData);
-  //     //         s.complete();
-  //     // });
-  //     // return s;
-  //     // if (!this.contextParameters.tab) {
-  //     //     this.contextParameters.tab = {};
-  //     // }
-  //     // this.contextParameters.tab.title = this.component.title;
-  //     // console.log(this.contextParameters.tab);
-  //     // this.cd.detectChanges();
-  //     // console.log(this.component.title);
-  //     // console.log(this.tabContent);
-  //     // console.log(this.contextParameters);
-  //     // this.contextParameters);
-  // }
-  // ngOnInit() {
-  //     this.contextParameters.tabTitle.next('test');
-  //     this.cd.detectChanges();
-  //     // super.fetchData();
-  // }
+  activatedElements: HTMLElement[] = [];
+
+  @ViewChild('header')
```

Comment:
I believe this is no longer needed

Reply:
oh, I assume this is causing the first tab to be activated when you load the page. Since this is in theory a list of elements, we should move to `ViewChildren` instead, and then we can use the H3 selector.

Reply:
I tried using the ViewChildren. The issue is you still need the ref (#header) to select it and I read (also tried it) that the elements are only usable in the `AfterViewInit` and at this point the toggle wasn't working.